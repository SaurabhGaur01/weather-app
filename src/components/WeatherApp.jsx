import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import handleRetrieveWeatherInfo from '../thunks/handleRetrieveWeatherInfo';
import Content from '../components/Content/Content';
import { TEMPERATURE_IN_FAHRENHEIT, DEFAULT_LOCATION } from '../constants/apiInfo';

class WeatherApp extends React.Component{
    componentDidMount(){
        const { actionCallWeatherApi } = this.props;
        actionCallWeatherApi(TEMPERATURE_IN_FAHRENHEIT, DEFAULT_LOCATION);
    }
    
    render(){
        return (
            <Content />
        )
    }
}

WeatherApp.propTypes = {
    actionCallWeatherApi: PropTypes.func.isRequired,
};

const mapDispatchAsProps = dispatch =>{
    return {
        actionCallWeatherApi: (temperature, location) => dispatch(handleRetrieveWeatherInfo(temperature, location)),
    };
}

const hocChain = compose(
    connect(null, mapDispatchAsProps),
);

export { WeatherApp as TestableWeatherApp };

export default hocChain(WeatherApp);