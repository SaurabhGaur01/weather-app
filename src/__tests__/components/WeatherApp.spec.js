import React from 'react';
import { shallow } from 'enzyme';
import { TestableWeatherApp } from '../../components/WeatherApp';

describe('WeatherApp Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionCallWeatherApi: jest.fn(),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableWeatherApp {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('componentDidMount', () => {
        it('should call actionCallWeatherApi', () => {
            const instance = shallow(<TestableWeatherApp {...props} />).instance();
            instance.componentDidMount();
            expect(props.actionCallWeatherApi).toHaveBeenCalled();
        })
    })
})