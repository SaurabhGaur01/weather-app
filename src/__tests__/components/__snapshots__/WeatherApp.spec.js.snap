// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherApp Component should render correctly 1`] = `<[object Object] />`;
